### PR TITLE
Support Tensor input for row conv op

### PR DESCRIPTION
**Pull request by LDOUBLEV**
Support tensor input for row_conv_op.  The input of row_conv_op can be tensor or lodtensor, and no other extra input is used.  When input is tensor, output is also tensor with padding.

### CODE REVIEW

**Review by qingqing01 — APPROVED** [review]
Has reviewed in https://github.com/PaddlePaddle/Paddle/pull/19296

File: paddle/fluid/operators/row_conv_op.cc [hunk 1]
```diff
@@ -1,5 +1,4 @@
 /* Copyright (c) 2016 PaddlePaddle Authors. All Rights Reserved.
-
```
Comment (heavengate) [inline paddle/fluid/operators/row_conv_op.cc:2]: 这行不用删

**Review by Superjomn — APPROVED** [review]
LGTM

File: paddle/fluid/operators/row_conv_op.cu [hunk 2]
```diff
@@ -348,9 +369,27 @@ class RowConvGradKernel<platform::CUDADeviceContext, T>
...
+    if (is_tensor) {
+      for (int i = 0; i < batch_size + 1; i++) {
+        batch_indices[i] = i * timesteps;
+      }
+      input_dim = X->dims()[2];
+    } else {
+      batch_indices = X->lod()[0];
+      input_dim = X->dims()[1];
+    }
+    // int input_dim = X->dims()[1];
```
Comment (heavengate) [inline paddle/fluid/operators/row_conv_op.cu:392]: detele

**Review by frankwhzhang — APPROVED** [review]
LGTM API.

File: paddle/fluid/operators/row_conv_op.cc (hunk 1 above)
Comment (LDOUBLEV) [reply paddle/fluid/operators/row_conv_op.cc:2]: OK

**Pull request merged**